### ISSUE TITLE
Invalid Commit References on Submodules

### ISSUE THREAD

**Issue by javaes** [mauer/xmidictrl#3]
Hi,

I was trying to build the v0.50 tag on my Linux machine, but git was not able to checkout out the sub-modules in `lib` because they were not found (all of them). The GitHub interface can also not find the commits, so I guess they are from some private branch.

I was able to build that project on my machine using [this commit ](https://github.com/mauer/xmidictrl/commit/8052d089f3cf00d040685d1dbf1d6a967808c34f), so I think building the latest version will be straight forward once the references are fixed.

Thanks for the help and setting up this project! :)

**Comment by mauer** [comment]
Hi,

Many thanks for your report. Looks like I messed up something with those annoying submodules. I will probably switch to subtree. It's easier to use and doesn't cause too many headaches.

I will try to fix it today or tomorrow.

Thanks,
Marco

**Comment by javaes** [comment]
Thanks! I got to build it by manually checking out the submodules in the meanwhile and will do my first flight tests soon :)

**Comment by mauer** [comment]
Hi,

I changed a few things and just compiled in successfully in Ubuntu 20.04 LTS. Is the plugin working correctly in Linux? I still have to install X-Plane in Linux to test it, so would be interested if everything is working fine.

Thanks,
Marco

**Comment by javaes** [comment]
Hi,

I'm not at home until the end of the Christmas holidays, so I won't be able to provide feedback on the latest version. I will surely try it when I'm back home.

At home I used the `v0.50` tag and was able to compile it by checking out the dependency libs manually at a version I deemed appropriate. After removing some duplication errors during the `make` (could be related to my limited C toolchain knowledge), it built successfully and I'm happily using it in my sim.

Thanks a lot again, from my side this issue can be closed.

Best,
Jan
